src/file_browser/mod.rs

```diff
@@ -608,9 +608,19 @@ fn populate_tree_nodes(
             // In case of invalid symlinks, we cannot obtain metadata.
             continue;
         };
+
+        let (mimetype, err) = gio::content_type_guess(Some(&filename), &[]);
+        let mimetype = mimetype.replace("/", "-");
```

Comment:
`get_for_content_type` essentially does the same as this hack, but returns the final icon